contracts/utils/Panic.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+/**
+ * @dev Helper library for emitting standardized panic codes.
+ */
+// slither-disable-next-line unused-state
+library Panic {
+    uint256 internal constant ASSERTION_ERROR = 0x1;
+    uint256 internal constant ARITHMETIC_UNDER_OR_OVERFLOW = 0x11;
+    uint256 internal constant DIVISION_BY_ZERO = 0x12;
+    uint256 internal constant ENUM_CONVERSION_OUT_OF_BOUNDS = 0x21;
+    uint256 internal constant INCORRECTLY_ENCODED_STORAGE_BYTE_ARRAY = 0x22;
+    uint256 internal constant POP_ON_EMPTY_ARRAY = 0x31;
+    uint256 internal constant ARRAY_ACCESS_OUT_OF_BOUNDS = 0x32;
+    uint256 internal constant TOO_MUCH_MEMORY_ALLOCATED = 0x41;
+    uint256 internal constant ZERO_INITIALIZED_VARIABLE = 0x51;
```

Comment:
Let's document these, as their meaning might not be obvious to users using the library without knowing these are already defined by Solidity.

Also, let's consider the original names given in the [Solidity repo](https://github.com/ethereum/solidity/blob/4577aebfd24a8dec47f7ce085d4e8887f5b8b3af/libsolutil/ErrorCodes.h#L36).

```suggestion
    /// @dev Called `assert` with an argument that evaluates to `false`.
    uint256 internal constant ASSERT = 0x1;
    /// @dev An arithmetic operation resulted in underflow or overflow.
    uint256 internal constant UNDER_OVERFLOW = 0x11;
    /// @dev Division or modulo operation by 0 (i.e. `x / 0` or `x % 0`).
    uint256 internal constant DIVISION_BY_ZERO = 0x12;
    /// @dev A number greater than 255 or negative was wrapped in an `enum`.
    uint256 internal constant ENUM_CONVERSION_ERROR = 0x21;
    /// ???
    uint256 internal constant STORAGE_ENCODING_ERROR = 0x22;
    /// @dev Attempted to `pop` an element out of an empty array.
    uint256 internal constant EMPTY_ARRAY_POP = 0x31;
    /// @dev Attempted to access an out-of-bound index at an empty array
    ///  (i.e. `x[i]` where `i >= x.length` or `i < 0`).
    uint256 internal constant ARRAY_OUT_OF_BOUNDS = 0x32;
    /// @dev If you allocate too much memory or create an array that is too large.
    uint256 internal constant RESOURCE_ERROR = 0x41;
    /// @dev Called an uninitialized internal function pointer.
    uint256 internal constant INVALID_INTERNAL_FUNCTION = 0x51;
```

Note I don't know how to trigger the `0x22` code. Also, I couldn't find what "too large" means in code `0x41`

Reply:
I used the name for hardhat helpers, which is handy for testing. I'm open to using Solidity's names, but then testing will be a bit more confusing.